### PR TITLE
[terra-table] Update guides and examples to use realistic data

### LINKED ISSUE #3057: [terra-table] Update table's guide data & examples to be more realistic
# Feature Request
Update tables guides' data & examples to be more emblematic of usage expectations.

## Description

All Guides:
Pick new table data type (i.e. fruit, cars, appliances, etc).

Selectable Guides:
Update row labels to indicate the expected toggle action.

Check Mark Table Guide:
Fully implement the 'discloseAction' will help consumers implement it closer to expectations.

## @ Mentions
@scottwilmarth @neilpfeiffer

### PR DESCRIPTION
**Pull request by saket2403**
### Summary
Updated guides and examples to use realistic data like cars, flatware, person details, etc.

Closes #3057

### Deployment Link

https://terra-core-deployed-pr-#.herokuapp.com/

### Additional Details

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-table/src/terra-dev-site/doc/guides/MultiSelectTable.jsx [hunk 1]
```diff
@@ -50,10 +50,10 @@ const MultiSelectTable = () => {
           checkLabel: 'Multi Selection',
         },
         cells: [
-          { key: 'cell-0', id: 'toggle-0', children: 'Column 0' },
-          { key: 'cell-1', id: 'toggle-1', children: 'Column 1' },
-          { key: 'cell-2', id: 'toggle-2', children: 'Column 2' },
-          { key: 'cell-3', id: 'toggle-3', children: 'Column 3' },
+          { key: 'cell-0', id: 'toggle-0', children: 'Name' },
+          { key: 'cell-1', id: 'toggle-1', children: 'Address' },
+          { key: 'cell-2', id: 'toggle-2', children: 'Phone  Number' },
```
Comment (avinashg1994) [inline packages/terra-table/src/terra-dev-site/doc/guides/MultiSelectTable.jsx:55]: nit: Extra space between "Phone" and "Number" should be removed.

File: packages/terra-table/src/terra-dev-site/doc/guides/mock-data/mock-select.js [hunk 2]
```diff
@@ -103,20 +103,20 @@ const mockData = [
     cells: [
       {
         key: 'unique-4-0',
-        title: 'Item 4-0',
+        title: 'Paul Smith',
       },
       {
```
Comment (avinashg1994) [inline packages/terra-table/src/terra-dev-site/doc/guides/mock-data/mock-select.js:109]: nit: Extra line can be removed

**Review by avinashg1994 — APPROVED** [review]

File: packages/terra-table/src/terra-dev-site/test/table/CheckMarkDemo.test.jsx [hunk 3]
```diff
@@ -79,10 +79,10 @@ const CheckMarkTable = () => {
...
         },
         cells: [
-          { key: 'cell-0', id: 'toggle-0', children: 'Column 0' },
-          { key: 'cell-1', id: 'toggle-1', children: 'Column 1' },
-          { key: 'cell-2', id: 'toggle-2', children: 'Column 2' },
-          { key: 'cell-3', id: 'toggle-3', children: 'Column 3' },
+          { key: 'cell-0', id: 'toggle-0', children: 'Name' },
+          { key: 'cell-1', id: 'toggle-1', children: 'Address' },
+          { key: 'cell-2', id: 'toggle-2', children: 'Phone Number' },
+          { key: 'cell-3', id: 'toggle-3', children: 'Symptom' },
```
Comment (emilyrohrbough) [inline packages/terra-table/src/terra-dev-site/test/table/CheckMarkDemo.test.jsx:85]:
```suggestion
          { key: 'cell-3', id: 'toggle-3', children: 'Email' },
```

File: packages/terra-table/src/terra-dev-site/doc/table/guides.5/SingleSelectTable.1.doc.mdx [hunk 4]
```diff
@@ -93,10 +93,10 @@ Finally we can implement a method to loop through our data and create the table
 +         checkLabel: 'Single Selection',
 +       },
 +       cells: [
```
Comment (emilyrohrbough) [inline packages/terra-table/src/terra-dev-site/doc/table/guides.5/SingleSelectTable.1.doc.mdx:95]: Should the summary have been updated?

File: packages/terra-table/src/terra-dev-site/doc/guides/MultiSelectTable.jsx (hunk 1 above)
Comment (saket2403) [reply packages/terra-table/src/terra-dev-site/doc/guides/MultiSelectTable.jsx]: Removed

File: packages/terra-table/src/terra-dev-site/doc/guides/mock-data/mock-select.js (hunk 2 above)
Comment (saket2403) [reply packages/terra-table/src/terra-dev-site/doc/guides/mock-data/mock-select.js]: Removed

File: packages/terra-table/src/terra-dev-site/test/table/CheckMarkDemo.test.jsx (hunk 3 above)
Comment (saket2403) [reply packages/terra-table/src/terra-dev-site/test/table/CheckMarkDemo.test.jsx]: Updated

File: packages/terra-table/src/terra-dev-site/doc/table/guides.5/SingleSelectTable.1.doc.mdx (hunk 4 above)
Comment (saket2403) [reply packages/terra-table/src/terra-dev-site/doc/table/guides.5/SingleSelectTable.1.doc.mdx:95]: Currently summary gives interaction details about table so left it as it is.

**Review by supreethmr — APPROVED** [review]

**Review by tbiethman — APPROVED** [review]

**Pull request merged**